pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -467,18 +469,18 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloud.MachineCreateDe
 	awsInstance := &awsInstance{instance: runOut.Instances[0]}
 
 	// Change to our security group
-	_, err = ec2Client.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
+	_, modifyInstanceErr := ec2Client.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
 		InstanceId: runOut.Instances[0].InstanceId,
 		Groups:     aws.StringSlice(config.SecurityGroupIDs),
 	})
-	if err != nil {
+	if modifyInstanceErr != nil {
 		_, err := ec2Client.TerminateInstances(&ec2.TerminateInstancesInput{
 			InstanceIds: []*string{runOut.Instances[0].InstanceId},
 		})
 		if err != nil {
-			return nil, awsErrorToTerminalError(err, fmt.Sprintf("failed to attach instance %s to security group id's %v & delete the created instance", aws.StringValue(runOut.Instances[0].InstanceId), config.SecurityGroupIDs))
+			return nil, awsErrorToTerminalError(err, fmt.Sprintf("failed to attach instance %s to security group id's %v due to %v & delete the created instance", aws.StringValue(runOut.Instances[0].InstanceId), config.SecurityGroupIDs, modifyInstanceErr))
```

Comment:
Could you reword this to `failed to delete instance %s due to %v after attaching to security groups %v failed due to %v`? 
Right now I find the error message a little confusing

Reply:
how about  `failed to delete instance %s due to %v after attaching to security groups %v which failed due to %v` ?

Reply:
SGTM